module/VuFind/src/VuFind/Auth/ILS.php

```diff
@@ -113,6 +113,23 @@ public function authenticate($request)
         return $this->handleLogin($username, $password, $loginMethod, $rememberMe);
     }
 
+    /**
+     * Does this authentication method support password recovery
+     *
+     * @return bool
+     */
+    public function supportsPasswordRecovery()
+    {
+        $driver = $this->getCatalog()->getDriver();
+        if (
+            method_exists($driver, 'changePassword')
```

Comment:
I'd suggest a new method for updating the password during recovery, since changePassword requires the patron from patronLogin method. Our custom code uses this:
https://github.com/NatLibFi/NDL-VuFind2/blob/dev/module/Finna/src/Finna/ILS/Driver/KohaRest.php#L792

Reply:
I've updated the code to use a resetPassword method for this. I've added my resetPassword to the SierraRest Driver.